Makefile

```diff
@@ -42,15 +42,20 @@ schema:
 	dune runtest ocaml/idl
 
 doc:
-	dune build --profile=$(PROFILE) ocaml/idl/datamodel_main.exe
+#html
 	dune build --profile=$(PROFILE) -f @ocaml/doc/jsapigen
 	mkdir -p $(XAPIDOC)/html
 	cp -r _build/default/ocaml/doc/api $(XAPIDOC)/html
 	cp _build/default/ocaml/doc/branding.js $(XAPIDOC)/html
 	cp ocaml/doc/*.js ocaml/doc/*.html ocaml/doc/*.css $(XAPIDOC)/html
-	dune exec --profile=$(PROFILE) -- ocaml/idl/datamodel_main.exe -closed -markdown $(XAPIDOC)/markdown
-	cp ocaml/doc/*.dot ocaml/doc/doc-convert.sh $(XAPIDOC)
+#markdown
+	dune build --profile=$(PROFILE) -f @ocaml/idl/markdowngen
+	mkdir -p $(XAPIDOC)/markdown
+	cp -r _build/default/ocaml/idl/autogen/*.md $(XAPIDOC)/markdown
+	cp -r _build/default/ocaml/idl/autogen/*.yml $(XAPIDOC)/markdown
 	find ocaml/doc -name "*.md" -not -name "README.md" -exec cp {} $(XAPIDOC)/markdown/ \;
+#other
+	cp ocaml/doc/*.dot ocaml/doc/doc-convert.sh $(XAPIDOC)
```

Comment:
```suggestion
	cp ocaml/doc/*.dot ocaml/doc/doc-convert.sh $(XAPIDOC)
	cd _build/install/default/xapi/doc; ./doc-convert.sh
```

Reply:
The output of this PR is in the xapi-doc rpm. Classes.png was never generated automatically. In the past this repo contained the pngs, but they were generated by running the script manually on demand, it was not part of the (make) build. Therefore I removed the pngs and left just the dot files; now the build just packages the dot files in the rpm, and the pngs are created by running the script in the pipeline that packages the API reference for delivery to the main product documentation.  If changes are needed in this workflow I'd rather make them in a different PR - I'm keen on making only the necessary changes needed to merge this one because there is further work depending on it.